src/python_testing/conformance_support.py

```diff
@@ -50,13 +50,50 @@ def __str__(self):
         return f"ConformanceException({self.msg})"
 
 
+class ChoiceConformanceException(ConformanceException):
+    def __str__(self):
+        return f'ChoiceExceptions({self.msg})'
+
+
+class BasicConformanceException(ConformanceException):
+    pass
+
+
+@dataclass(frozen=True)
+class Choice:
+    choice: str
+    more: bool
+
+    def __str__(self):
+        more_str = '+' if self.more else ''
+        return '.' + self.choice + more_str
+
+
+def parse_choice(element: ElementTree.Element) -> Optional[Choice]:
+    choice = element.get('choice', '')
```

Comment:
nit: if you do a `if not choice: return None` you can rely that choice is valid in the lines below (saves some conditions)`.

is `more` a valid field if choice is not specified?